p2p/peer.go

```diff
@@ -300,6 +298,16 @@ func (p *peer) IsPersistent() bool {
 	return p.peerConn.persistent
 }
 
+// IPHasChanged returns true and the new IP if the peer's IP has changed.
+func (p *peer) IPHasChanged() bool {
```

Comment:
```suggestion
func (p *peer) HasIPChanged() bool {
```
nit: but this sounds better